packages/cadl-rlc-emitter/src/transform/transfromRLCOptions.ts

```diff
@@ -21,5 +21,51 @@ export function transformRLCOptions(program: Program): RLCOptions {
       config.endpoint = host[0].url;
     }
   }
+  const securityInfo = processAuth(program);
+  config.addCredentials = config.addCredentials !== false? securityInfo.addCredentials: false;
```

Comment:
securityInfo.addCredentials could be empty?
```suggestion
  config.addCredentials = config.addCredentials !== false? securityInfo?.addCredentials: false;
```